pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2385,6 +2414,9 @@ func (tmpl *Template) GetType() TemplateType {
 	if tmpl.Resource != nil {
 		return TemplateTypeResource
 	}
+	if tmpl.MulticlusterResource != nil {
+		return TemplateTypeMulticlusterResource
```

Comment:
how come this is needed?

Reply:
I was trying to give `Resource2` from your Poc a more descriptive name: https://github.com/argoproj/argo-workflows/pull/4781/files#diff-94d0f1284fafd0a1cb05dc2309b4be043b6fb01d66626ab1867d40c84b573ec2R541

`MulticlusterResource` seemed like it made sense. Do you have another suggestion? I'm not always the best at naming